zulipterminal/ui_tools/utils.py

```diff
@@ -30,6 +31,10 @@ def create_msg_box_list(
     last_msg = last_message
     muted_msgs = 0  # No of messages that are muted.
     for msg in message_list:
+        if model.initial_data.get("realm_enable_guest_user_indicator", None):
+            user_info = model.get_user_info(msg["sender_id"])
+            role = user_info["role"]
+            msg["sender_role"] = ROLE_BY_ID[role]["name"]
```

Comment:
I understand what you're trying to achieve here, ie. caching the name of the role in messages.

This does work, but the messages data structure already potentially has fields dedicated to various roles, as well as the main role field in late versions.

Reply:
so ill make a separate method which will make the two parts of the name the name and suffix 
by default the suffix will be "" (blank)  so that way we do not have to check any setting from model and wherever we are displaying names we will call this method ? does this work ?

Reply:
Yes, that may be useful, though it may depend on how the UI works in different places, eg. our custom buttons do not accept the same input as a simple text box. If you apply it to one area first, then you can just have the logic at that position.

The logic may be quite small, once you have the new method in the model, to indicate if a guest suffix should be shown for a given user id.